pymc3/step_methods/smc.py

```diff
@@ -147,6 +147,10 @@ def __init__(self, vars=None, out_vars=None, n_chains=100, scaling=1., covarianc
         vars = inputvars(vars)
 
         if out_vars is None:
+            if not any(likelihood_name == RV.name for RV in model.unobserved_RVs):
```

Comment:
I think we should not leave the option for user to define Deterministic likelihood.